lib/parfor.h

```diff
@@ -24,6 +24,20 @@ namespace qsim {
 
 template <uint64_t MIN_SIZE>
 struct ParallelForT {
+  static bool CanBeParallel(uint64_t size) {
+    return size >= MIN_SIZE;
+  }
+
+  static uint64_t GetIndex0(
```

Comment:
To get a measured bitstring, we need to sum up the bitsting probabilities until the sum does not exceed a given random number. We can reuse partial sums obtained at the previous step when we calculated the norm using threads. First, we find the first cumulative partial sum that exceeds our random number. This can be done quickly as the number of threads is much smaller than the number of bit strings. Then we iterate over bitstring indices beginning with the first index and ending with the last index that were used to obtain the required partial sum. That is we need to know exactly the same indices that were used by `ParallelFor` to divide work between threads.